src/main/java/org/opensearch/flowframework/transport/GetWorkflowRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.transport;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.Nullable;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Transport Request to get a workflow or workflow status
+ */
+public class GetWorkflowRequest extends ActionRequest {
+
+    /**
+     * The documentId of the workflow entry within the Global Context index
+     */
+    @Nullable
+    private String workflowId;
```

Comment:
why can workflowId be nullable?

Reply:
This is just for the createWorkflow dry run case. the dry run is executed as a validation before we index and before we have a workflow ID, previously I passed a randomly generated ID, I have that part commented out now, I was responding to comment by Dan to make it nullable if thats okay I will remove comment